AC3.2-MidUnit-6-Exam/DoubleVideoViewController.swift

```diff
@@ -7,25 +7,109 @@
 //
 
 import UIKit
+import AVKit
+import AVFoundation
+import MobileCoreServices
 
-class DoubleVideoViewController: UIViewController, CellTitled {
+class DoubleVideoViewController: UIViewController, CellTitled, UIImagePickerControllerDelegate, UINavigationControllerDelegate {
+    //MARK: - Properties
     var titleForCell: String = "Double Video"
-
+    var imagePickerController: UIImagePickerController!
+    var videoURL: URL?
+    var playerA: AVPlayer?
+    var playerB: AVPlayer?
+    
+    //MARK: - Outlets
     @IBOutlet weak var videoContainerTop: UIView!
     @IBOutlet weak var videoContainerBottom: UIView!
     
+    //MARK: - Methods
     override func viewDidLoad() {
         super.viewDidLoad()
-
+        
         // don't f with this line
         self.edgesForExtendedLayout = UIRectEdge(rawValue: 0)
     }
-
-    override func viewDidLayoutSubviews() {
-        // update your layers' frames here
+    
+    //    override func viewDidLayoutSubviews() {
+    //        // update your layers' frames here
+    //        //this isn't actually doing anything
+    //    }
+    
+    func handleVideoPlaying() {
+        if let url = self.videoURL {
+            let asset = AVAsset(url: url)
+            let playerItem = AVPlayerItem(asset: asset)
+            let player = AVPlayer(playerItem: playerItem)
+            let playerLayer = AVPlayerLayer(player: player)
+            
+            //self.player.rate == 0.0 did not lead to a positive response so I am successfully using self.player == nil
+            if self.playerA == nil {
+                //bonus
+                self.videoContainerTop.layer.sublayers?.forEach { (layer) in
+                    if layer == AVPlayerLayer() {
+                        layer.removeFromSuperlayer()
+                    }
+                }
+                
+                self.playerA = player
+                playerLayer.frame = self.videoContainerTop.bounds
+                self.videoContainerTop.layer.addSublayer(playerLayer)
+                
+                self.playerA?.playImmediately(atRate: 1.0)
+            }
+            else if self.playerB == nil {
+                self.videoContainerBottom.layer.sublayers?.forEach { (layer) in
+                    if layer == AVPlayerLayer() {
+                        layer.removeFromSuperlayer()
+                    }
+                }
+                
+                self.playerB = player
+                playerLayer.frame = self.videoContainerBottom.bounds
+                self.videoContainerBottom.layer.addSublayer(playerLayer)
+                
+                self.playerB?.playImmediately(atRate: 1.0)
+            }
+            else {
+                //bonus
+                let alertController = UIAlertController(title: "Both players are in use.", message: "Wait till one completes.", preferredStyle: .alert)
```

Comment:
After the first two views are filled this happens every time. You got partial bonus for the reporting but the detection would be more complicated when the viewers were reloading.